model/jpa/src/main/java/org/keycloak/storage/jpa/JpaUserFederatedStorageProvider.java

```diff
@@ -691,12 +691,12 @@ public CredentialModel getStoredCredentialById(RealmModel realm, UserModel user,
 
     @Override
     public List<CredentialModel> getStoredCredentials(RealmModel realm, UserModel user) {
-        return getStoredCredentials(realm, user.getId());
+        return getStoredCredentialsStream(realm, user.getId()).collect(Collectors.toList());
```

Comment:
The other way around - things in the federated package need to preserve the original collection-based methods.

The original PR that introduced the streams for User interfaces ended up changing the interfaces in the federated package (as they are tightly related) but it didn't touch the UserCredentialStore, this is why this particular method hasn't been updated yet. Once we streamify UserCredentialStore it will be replaced with the Stream-based implementation.

If you feel like UserCredentialStore should be changed as well as part of this work I can change this PR to include the its streamification and change everything at once.

Reply:
Update: for consistency I've decided to streamify the UserCredentialStore and UserCredentialManager interfaces as part of this work (see second commit in this PR).